dynamodb_utils/awslambda.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+
+import importlib
+import json
+import logging
+import os
+
+from pynamodb.connection import Connection
+
+from . import dumper
+from .multi import awslambda
+
+logger = logging.getLogger(__name__)
+
+def handler(event, context):
+    if not isinstance(event, dict):
+        logger.error('event not a dict')
+        return
+
+    source = event.get('source', None)
+
+    if source == dumper.__name__:
+        tables = []
+        table = event.get('table', '*')
+        dest = event.get('destination', None)
+        config = event.get('config', {'region': os.environ.get('AWS_REGION', None)})
+
+        if not dest:
+            logger.error('Destination not found in event!')
+            return
+
+        if table == '*':
+            try:
+                connection = Connection(host=config.get('host', None), region=config.get('region', None))
+                tables = connection.list_tables().get('TableNames', [])
+            except Exception as e:
+                logger.error('Failed to list tables to be backed up: {0}'.format(e))
+        else:
+            tables = re.split('\s+/', source)
+
+        if not tables:
+            logger.error('No tables to back up!')
+            return
+
+        for table in tables:
+            try:
+                logger.debug('Dumping table {0} to {1} with config {2}'.format(table, dest, config))
+                dumper.dump_table(table_name=table, destination=dest, **config)
+            except Exception as e:
+                logger.error('Failed to dump table {0} to {1}: {2}'.format(table, dest, e))
+
+    elif source == awslambda.__name__:
+        args = event.get('args', [])
+        target = event.get('target', {})
+        module = target.get('module', None)
+        name = target.get('name', None)
+
+        if module and name:
+            mod = importlib.import_module(module)
+            func = getattr(mod, name, None)
+            if callable(func):
+                logger.debug('Calling function: {0}'.format(func))
+                func(*args)
+            else:
+                logger.error('Selected function is not callable: {0}.{1}'.format(module, name))
+        else:
+            logger.error('Target module and/or name not found in event!')
```

Comment:
again probably too defensive and better to just do things like `event['module']` and `func(*args)`

stack traces provide more information than your debug messages :)